account_invoice_line_default_account/models/account_invoice_line.py

```diff
@@ -0,0 +1,58 @@
+# Copyright 2012 Therp BV (<http://therp.nl>)
+# Copyright 2013-2018 BCIM SPRL (<http://www.bcim.be>)
+# Copyright 2022 Simone Rubino - TAKOBI
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models, _
+
+
+class AccountInvoiceLine(models.Model):
+    _inherit = 'account.invoice.line'
+
+    def _account_id_default(self):
+        partner_id = self._context.get('partner_id')
+        if not partner_id:
+            return self._default_account()
+        assert isinstance(partner_id, int), (
+            _('No valid id for context partner_id %d') % partner_id)
+        invoice_type = self._context.get('type')
+        if invoice_type in ['in_invoice', 'in_refund']:
+            partner = self.env['res.partner'].browse(partner_id)
+            if partner.property_account_expense:
+                return partner.property_account_expense.id
+        elif invoice_type in ['out_invoice', 'out_refund']:
+            partner = self.env['res.partner'].browse(partner_id)
+            if partner.property_account_income:
+                return partner.property_account_income.id
+        return self._default_account()
+
+    account_id = fields.Many2one(default=_account_id_default)
+
+    @api.onchange('account_id')
+    def _onchange_account_id(self):
```

Comment:
maybe better to use a different name to not have to call super

Reply:
Thanks for the review!
I can see no harm in calling `super`, actually it looks to me like a common thing to call `super` when overriding a method; why should this case be treated in a different way?

Reply:
Before `api.onchange`, there could be only a single method called on an onchange. With `api.onchange` you can have multiple different methods. The conditions are not to test if super has to be called or not but if this method has to be executed. So I find the current implementation hard to read.
Nevertheless it will work. Just a minor remark.

Reply:
That makes perfect sense: overriding the method is really needed only when you want to edit the method's behavior, but that does not have to be the way to go with the `onchange` methods.
I have done this change too, with a bit of explanation in the comments, let me know if it looks better now :)

Reply:
Looks clear now